root/static/scripts/edit/URLCleanup.js

```diff
@@ -536,15 +619,25 @@ const CLEANUPS = {
   },
   secondhandsongs: {
     match: [new RegExp("^(https?://)?([^/]+\\.)?secondhandsongs\\.com/", "i")],
-    type: LINK_TYPES.secondhandsongs
+    type: LINK_TYPES.secondhandsongs,
+    validate: function (url, id) {
+      switch (id) {
+        case LINK_TYPES.secondhandsongs.artist:
+          return /secondhandsongs\.com\/artist\//.test(url);
+        case LINK_TYPES.secondhandsongs.release:
+          return /secondhandsongs\.com\/release\//.test(url);
+        case LINK_TYPES.secondhandsongs.work:
+          return /secondhandsongs\.com\/work\//.test(url);
+      }
+      return true;
```

Comment:
A `validate` function should never return `true` because it allows to register any matched URL to any restricted link type such as `LINK_TYPE.amazon.release`. It should instead allow unrestricted link type only with `RESTRICTED_LINK_TYPES.indexOf(id) === -1`, or preferably we could add a function `isUnrestricted` that computes it and have `validate` functions return `isUnrestricted(id)` rather than `true`. This applies to `bandcamp`, `songkick`, and `setlistfm` too.